addons/sale_timesheet/models/account.py

```diff
@@ -94,7 +94,7 @@ def _timesheet_preprocess(self, values):
             task = self.env['project.task'].browse(values.get('task_id'))
             if task.analytic_account_id:
                 values['account_id'] = task.analytic_account_id.id
-                values['company_id'] = task.analytic_account_id.company_id.id
+                values['company_id'] = task.analytic_account_id.company_id.id or task.company_id.id
```

Comment:
Thanks for the fix 🙂 Nice PR with steps to reproduce, description of the problem and solution.
Just a very nitpicking thing:
When you return from the line in order to prevent too long text, it is better to keep the line indent to increase readability in my opinion.
```txt
3. Go to Settings > Accounting > Analytics and enable
Analytic Accounting
```
would become
```
3. Go to Settings > Accounting > Analytics and enable
   Analytic Accounting
```